x-pack/legacy/plugins/siem/server/index.ts

```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { PluginInitializerContext } from 'src/core/server';
+import { Plugin } from './plugin';
+
+export const plugin = (context: Pick<PluginInitializerContext, 'logger'>) => {
```

Comment:
Nothing wrong with doing so, but when you're using any NP types directly from `src/core/server` there's no need to pick your dependencies from it. We recommend this when you're consuming from legacy, so that you're very sure what legacy dependencies you have left.

Reply:
You're right, adding a type assertion on the current shim invocation is a much better solution 👍

Reply:
7cc80aa